### DISCUSSION TITLE
How to get window size to scale panels accordingly.

### DISCUSSION THREAD

**Discussion by rik-rosseel** [emilk/egui#2015] [Q&A] [answered]
I would like to make an app with a side panel and a main panel. I'm trying to get the side panel to be a percentage of the entire window (e.g.: 20% of window width). How would I be able to get the window size to make the sidepanel bigger or smaller if the window is resized?

**Comment by rik-rosseel** [comment, marked as answer]
After some more digging not knowing where to look I stumbled on the answer: The winow size is under `frame.info().window_info.size`.

**Reply by pauladam94** [reply]
What is the answer with the newer version of egui an eframe ?

**Reply by wangxiaochuTHU** [reply]
try out `ctx.input(|i| i.viewport().outer_rect)`?

**Reply by AravDesai** [reply]
I was able to get the size of the window with `ctx.input(|i: &egui::InputState| i.screen_rect())`

**Reply by anderspitman** [reply]
@emilk is there any advantage to `ctx.input(|i: &egui::InputState| i.screen_rect())` vs just `ctx.screen_rect()`?
